go-controller/pkg/cni/helper_linux.go

```diff
@@ -354,6 +354,12 @@ func ConfigureOVS(ctx context.Context, namespace, podName, hostIfaceName string,
 		ifaceID = util.GetSecondaryNetworkIfaceId(namespace, podName, ifInfo.NADName)
 	}
 	initialPodUID := ifInfo.PodUID
+	pod, err := PodInfoGetter.getPod(getter, namespace, podName)
+	if err != nil {
+		klog.Warningf("Could not find pod %v %v", pod, err)
+	} else if IsStaticPod(pod) {
+		initialPodUID = string(pod.ObjectMeta.UID)
+	}
```

Comment:
I don't think we need this change. 

`ifInfo.PodUID` should already be set to the correct pod UID:
In `checkOrUpdatePodUID` we set the pod UID in the PorRequest, then in  PodAnnotation2PodInfo is set to PodInterfaceInfo which we are using here.

Reply:
we hit checkOrUpdatePodUID before we set the correct UID. When I remove the static pod check from the function, the static pod will not create with the pd deleted before sandbox op began error, so at this point the uid has not been updated.

Reply:
you are right, removed the change